File: examples/noise/noise.js
function setup(c) {
  c.savedCanvasFontSize(24);
  c.cellPaddingDouble();
}

function draw(c) {
  for (let i = 0; i < c.width; i++) {
    for (let j = 0; j < c.height; j++) {
      let color = c.map(
        c.noise2D(
          i * 0.009 + c.framecount / 1000,
          j * 0.009 + c.framecount / 1000,
        ),
        0,
        1,
        150,
        231,
      );
      c.stroke("§", color, "#000000");
      c.point(i, j);
    }
  }
}

function onKey(c, e) {
  if (e.Key == "c") {
    c.saveCanvasToPNG(`canvas_${Date.now()}.png`);
  }
  if (e.key == "space") {
    c.noiseSeed(Date.now());
    c.redraw();
  }
}

function onMouse(c, e) {
  if (e.type == "click") {
    c.noiseSeed(Date.now());
    c.redraw();
  }
}

File: examples/noise/noise.go
package main

import (
	"context"
	"fmt"
	"strconv"
	"time"

	"github.com/emprcl/runal"
)

func main() {
	runal.Run(context.Background(), setup, draw, runal.WithOnKey(onKey))
}

func setup(c *runal.Canvas) {
	c.SavedCanvasFontSize(24)
	c.CellPaddingDouble()
}

func draw(c *runal.Canvas) {
	for i := 0; i < c.Width; i++ {
		for j := 0; j < c.Height; j++ {
			color := c.Map(
				c.Noise2D(
					float64(i)*0.009+float64(c.Framecount)/1000,
					float64(j)*0.009+float64(c.Framecount)/1000,
				),
				0,
				1,
				150,
				231,
			)
			colorStr := strconv.FormatFloat(color, 'f', -1, 64)
			c.Stroke("§", colorStr, "#000000")
			c.Point(i, j)
		}
	}
}

func onKey(c *runal.Canvas, e runal.KeyEvent) {
	if e.Key == "c" {
		filename := fmt.Sprintf("canvas_%d.png", time.Now().Unix())
		c.SaveCanvasToPNG(filename)
	}
	if e.Key == "space" {
		c.NoiseSeed(time.Now().Unix())
		c.Redraw()
	}
}

func onMouseClick(c *runal.Canvas, e runal.MouseEvent) {
	c.NoiseSeed(time.Now().Unix())
	c.Redraw()
}
